src/main/scala/util/ClockGate.scala

```diff
@@ -3,17 +3,29 @@
 package freechips.rocketchip.util
 
 import chisel3._
+import chisel3.util.{HasBlackBoxResource, HasBlackBoxPath}
 import freechips.rocketchip.config.{Field, Parameters}
 
+import java.nio.file.{Files, Paths}
+
 case object ClockGateImpl extends Field[() => ClockGate](() => new EICG_wrapper)
+case object ClockGateModelFile extends Field[Option[String]](None)
 
-abstract class ClockGate extends BlackBox {
+abstract class ClockGate extends BlackBox
+  with HasBlackBoxResource with HasBlackBoxPath {
   val io = IO(new Bundle{
     val in = Input(Clock())
     val test_en = Input(Bool())
     val en = Input(Bool())
     val out = Output(Clock())
   })
+
+  def addVerilogResource(vsrc: String): Unit = {
+    if (Files.exists(Paths.get(vsrc)))
```

Comment:
Is the a reason for this to not just error immediately if the specified file does not exist?
The fallback might be unexpected